src/StackExchange.Redis/Interfaces/IDatabase.cs

```diff
@@ -325,6 +325,153 @@ public interface IDatabase : IRedis, IDatabaseAsync
+        /// <param name="hashField">The field in the hash to set expire time.</param>
+        /// <param name="expiry">The exact date to expiry to set.</param>
+        /// <param name="when">under which condition the expiration will be set using <see cref="ExpireWhen"/>.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns> null if the key does not exist. Otherwise returns the result of operation for given field;
+        ///		2: field deleted because the specified expiration time is due
+        ///		1: expiration time set/updated
+        /// 	0: expiration time is not set/update (a specified ExpireWhen condition is not met)
+        ///		-2 : no such field
+        /// </returns>
+        ExpireResult? HashFieldExpire(RedisKey key, RedisValue hashField, DateTime expiry, ExpireWhen when = ExpireWhen.Always, CommandFlags flags = CommandFlags.None);
+
+        /// <summary>
+        ///	Set the time out on a field of the given set of fields of hash
+        /// After the timeout has expired, the field of the hash will automatically be deleted.
+        /// </summary>
+        /// <param name="key">The key of the hash.</param>
+        /// <param name="hashFields">The fields in the hash to set expire time.</param>
+        /// <param name="expiry">The exact date to expiry to set.</param>
+        /// <param name="when">under which condition the expiration will be set using <see cref="ExpireWhen"/>.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns> null if the key does not exist. Otherwise returns an array where each item is the result of operation for given fields;
+        ///		2: field deleted because the specified expiration time is due
+        ///		1: expiration time set/updated
+        /// 	0: expiration time is not set/update (a specified ExpireWhen condition is not met)
+        ///		-2 : no such field
+        /// </returns>
+        ExpireResult[]? HashFieldExpire(RedisKey key, RedisValue[] hashFields, DateTime expiry, ExpireWhen when = ExpireWhen.Always, CommandFlags flags = CommandFlags.None);
+
+        /// <summary>
+        /// For specified field, it gets the expiration time as a Unix timestamp in milliseconds (milliseconds since the Unix epoch)
+        /// </summary>
+        /// <param name="key">The key of the hash.</param>
+        /// <param name="hashField">The field in the hash to get expire time.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>null if the key does not exist. Otherwise returns the result of operation for given field
+        ///     expiration time: as a UNIX timestamp in milliseconds
+        ///     -1: if field has no associated expiration time 
+        ///     -2: no such field
+        /// </returns>
+        long? HashFieldExpireTime(RedisKey key, RedisValue hashField, CommandFlags flags = CommandFlags.None);
```

Comment:
This should be DateTime? presumably? Namewise, this sounds like an action rather than a query - something "get expiry" perhaps? Also comparable to KeyTimeToLive (although that's the "TimeSpan" rather than "DateTime" approach)

Reply:
fixed